src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4212,8 +4195,8 @@ private void CheckEqualArgTypes(TexlNode left, TexlNode right)
                 if (!(typeLeft.IsPrimitive && typeRight.IsPrimitive) && !(typeLeft.IsPolymorphic && typeRight.IsPolymorphic) && !(typeLeft.IsControl && typeRight.IsControl)
                     && !(typeLeft.IsPolymorphic && typeRight.IsRecord) && !(typeLeft.IsRecord && typeRight.IsPolymorphic))
                 {
-                    var leftTypeDisambiguation = typeLeft.IsOptionSet && typeLeft.OptionSetInfo != null ? $"({typeLeft.OptionSetInfo.Name})" : string.Empty;
-                    var rightTypeDisambiguation = typeRight.IsOptionSet && typeRight.OptionSetInfo != null ? $"({typeRight.OptionSetInfo.Name})" : string.Empty;
+                    var leftTypeDisambiguation = typeLeft.IsOptionSet && typeLeft.OptionSetInfo != null ? $"({typeLeft.OptionSetInfo.EntityName})" : string.Empty;
```

Comment:
This is for displaying in an error message. Using the entity name which is what Authors would use to reference this in formulas instead of an internal CDS-specific `Name` field that may not show up to the user will result in a better error message, an accidental benefit of this change.